modules/programs/zellij.nix

```diff
@@ -39,6 +39,30 @@ in {
         list of options.
       '';
     };
+
+    enableBashIntegration = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
```

Comment:
We usually make shell integration options default to `true`, using something like `mkEnableOption "Foo integration" // { default = true; }`.

Reply:
Ok, maybe `false` is better for backwards compatibility. I'd still go with `mkEnableOption`.

Reply:
Could the default be based on `stateVersion` (`false` if `stateVersion` < 23.05)?

Reply:
We can assume most configs that are enabling the zellij package will be configuring the shell integrations themselves. So if we enable these shell integrations as the default there will be some redundant code in their respective shells' rc files.

While I haven't tested it yet, I do think that these shell integration evals are idempotent, so it could be relatively safe to enable them by default. There could be the cases, though, where a users with more bespoke integrations encounter conflicts. Not sure how common it is to deviate from the documented approach, though.

So there is a decent case for either option, imo. I'm happy to go with whatever y'all prefer.

Reply:
Let's go with `true`.